### PR TITLE
[8.0] Allow to define global log levels

### PR DESCRIPTION
**Pull request by chaen**
Before this PR, it is basically impossible to say "I want all agents to run in VERBOSE".
Now you can :-) 

BEGINRELEASENOTES

* Framework
CHANGE: Default log level is INFO
NEW: Define global configuration options for log level

ENDRELEASENOTES

### CODE REVIEW

File: src/DIRAC/FrameworkSystem/private/standardLogging/LoggingRoot.py [hunk 1]
```diff
@@ -116,14 +116,45 @@ def initialize(self, systemName, cfgPath, forceInit=False):
...
+        * Operations/<setup/vo>/Logging/Default<Agents/Services>LogLevel
+        * Operations/<setup/vo>/Logging/DefaultLogLevel
+
+        :param str cfgPath: configuration path
+        """
+        # We have to put the import line here to avoid a dependancy loop
+        from DIRAC.ConfigurationSystem.Client.Helpers.Operations import Operations
+        from DIRAC import gConfig
+
+        # Search desired backends in the component
```
Comment (aldbr) [inline src/DIRAC/FrameworkSystem/private/standardLogging/LoggingRoot.py]:
```suggestion
        # Search desired loglevel in the component
```

**Review by aldbr — APPROVED** [review]

File: src/DIRAC/FrameworkSystem/private/standardLogging/LoggingRoot.py [hunk 2]
```diff
@@ -50,8 +50,8 @@ def __init__(self):
         for level in levels:
             logging.addLevelName(levels[level], level)
 
-        # root Logger level is set to NOTICE by default
-        self._logger.setLevel(LogLevels.NOTICE)
+        # root Logger level is set to INFO by default
+        self._logger.setLevel(LogLevels.INFO)
```
Comment (fstagni) [inline src/DIRAC/FrameworkSystem/private/standardLogging/LoggingRoot.py:54]: I understand why you do this, but I would not do it in a patch release. 

I guess this is impacting the commands (scripts) right? There might be nasty effects for the Pilot that, remember, (horribly) parses the output of few DIRAC commands.

Comment (chaen) [reply src/DIRAC/FrameworkSystem/private/standardLogging/LoggingRoot.py:54]: I think a logging change is really not something that deserves waiting a major release.

If anything, we would see more in the scripts.

Where are we still parsing anything ? :-(

Comment (chrisburr) [reply src/DIRAC/FrameworkSystem/private/standardLogging/LoggingRoot.py:54]: I think it's just these four places:
* https://github.com/DIRACGrid/Pilot/blob/acd5754e041410e5bd99dd62f5c1c820b0186360/Pilot/pilotCommands.py#L611
* https://github.com/DIRACGrid/Pilot/blob/acd5754e041410e5bd99dd62f5c1c820b0186360/Pilot/pilotCommands.py#L538
* https://github.com/DIRACGrid/Pilot/blob/acd5754e041410e5bd99dd62f5c1c820b0186360/Pilot/pilotCommands.py#L946-L948
* https://github.com/DIRACGrid/Pilot/blob/acd5754e041410e5bd99dd62f5c1c820b0186360/Pilot/pilotCommands.py#L971-L975

Though I think we should target v8.0 as the pilot is more robust to this than it used to be and we can quickly fix it if there are any unexpected consiquences.

**Review by chrisburr — APPROVED** [review]

File: src/DIRAC/FrameworkSystem/private/standardLogging/LoggingRoot.py (hunk 2 above)
Comment (fstagni) [reply src/DIRAC/FrameworkSystem/private/standardLogging/LoggingRoot.py:54]: There are those places, yes. This was actually not difficult to test: https://jenkins-dirac.web.cern.ch/view/DIRAC/job/Pilot3_CVM4_pipeline/376/

I actually forgot that some time ago we added a sort of "protection" for these cases, so should be alright.

**Review by fstagni — APPROVED** [review]

**Pull request merged**